Incorrect wildcard bound determination with stub files

Take the following simple class:

```
class NonN<T> {}
```

Compile with normal javac.
Then add a file `NonN.astub` that specifies the upper bound is non-null:

```
import org.checkerframework.checker.nullness.qual.NonNull;

class NonN<T extends @NonNull Object> {}
```

Then add this main class:

```
class Wildcards {
  NonN<?> f = new NonN<Object>();

  class LocalNonN<T extends Object> {}
  LocalNonN<?> g = new LocalNonN<Object>();
}
```

Compile with `.../javac -processor nullness -Astubs=NonN.astub Wildcards.java` and you get:

```
Wildcards.java:2: error: [type.argument.type.incompatible] incompatible types in type argument.
  NonN<?> f = new NonN<Object>();
       ^
  found   : ? extends @Initialized @Nullable Object
  required: @Initialized @NonNull Object
1 error
```

The behavior for the wildcard is different for `NonN` and `LocalNonN`, which is wrong.
I was able to reproduce it with this stub-file setup, but maybe there is a simpler way to reproduce this.

--- a/checker/jtreg/stubs/wildcards/NonN.astub
+++ b/checker/jtreg/stubs/wildcards/NonN.astub
@@ -1,0 +1,4 @@
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+class NonN<T extends @NonNull Object> {
+}

--- a/checker/jtreg/stubs/wildcards/NonN.java
+++ b/checker/jtreg/stubs/wildcards/NonN.java
@@ -1,0 +1,1 @@
+class NonN<T> {}

--- a/checker/jtreg/stubs/wildcards/Wildcards.java
+++ b/checker/jtreg/stubs/wildcards/Wildcards.java
@@ -1,0 +1,15 @@
+/*
+ * @test
+ * @summary Test for Issue #1055
+ * @library .
+ * @compile NonN.java
+ * @compile -XDrawDiagnostics -processor org.checkerframework.checker.nullness.NullnessChecker -Astubs=NonN.astub Wildcards.java
+ */
+
+class Wildcards {
+    NonN<?> f = new NonN<Object>();
+
+    class LocalNonN<T extends Object> {}
+
+    LocalNonN<?> g = new LocalNonN<Object>();
+}

--- a/framework/src/org/checkerframework/framework/stub/StubParser.java
+++ b/framework/src/org/checkerframework/framework/stub/StubParser.java
@@ -501,7 +501,7 @@ public class StubParser {
             */
         }
 
-        annotateTypeParameters(typeArguments, typeParameters);
+        annotateTypeParameters(atypes, typeArguments, typeParameters);
         annotateSupertypes(decl, type);
         putNew(atypes, elt, type);
         List<AnnotatedTypeVariable> typeVariables = new ArrayList<>();
@@ -568,7 +568,7 @@ public class StubParser {
         addDeclAnnotations(declAnnos, elt);
 
         AnnotatedExecutableType methodType = atypeFactory.fromElement(elt);
-        annotateTypeParameters(methodType.getTypeVariables(), decl.getTypeParameters());
+        annotateTypeParameters(atypes, methodType.getTypeVariables(), decl.getTypeParameters());
         typeParameters.addAll(methodType.getTypeVariables());
         annotate(methodType.getReturnType(), decl.getType());
 
@@ -846,7 +846,9 @@ public class StubParser {
     }
 
     private void annotateTypeParameters(
-            List<? extends AnnotatedTypeMirror> typeArguments, List<TypeParameter> typeParameters) {
+            Map<Element, AnnotatedTypeMirror> atypes,
+            List<? extends AnnotatedTypeMirror> typeArguments,
+            List<TypeParameter> typeParameters) {
         if (typeParameters == null) {
             return;
         }
@@ -875,6 +877,7 @@ public class StubParser {
                     stubWarnIfNotFound("Annotations on intersection types are not yet supported");
                 }
             }
+            putNew(atypes, paramType.getUnderlyingType().asElement(), paramType);
         }
     }
 

